### PR TITLE
windows 2019 images should have curl.exe by default

### PR DESCRIPTION
**Pull request by jetersen**
This will allow windows to also run deploy

### CODE REVIEW

File: vars/infra.groovy [hunk 1]
```diff
@@ -259,11 +259,17 @@ void prepareToPublishIncrementals() {
 void maybePublishIncrementals() {
     if (isRunningOnJenkinsInfra() && currentBuild.currentResult == 'SUCCESS') {
         stage('Deploy') {
-            node('maven || linux') {
+            node('maven || linux || windows') {
```
Comment (jetersen) [inline vars/infra.groovy:262]: Could potentially drop the label spec if all windows nodes are using 2019 base.

**Comment by olblak** [comment]
If curl.exe is not already available on windows machine, feel free to add on the packer images. [Link](https://github.com/jenkins-infra/packer-images/)

**Comment by jetersen** [comment]
Nah it should already be there, Microsoft put it in :)

https://devblogs.microsoft.com/commandline/tar-and-curl-come-to-windows/

This includes server and even docker images.

File: vars/infra.groovy [hunk 2]
```diff
@@ -259,11 +259,17 @@ void prepareToPublishIncrementals() {
...
                 withCredentials([string(credentialsId: 'incrementals-publisher-token', variable: 'FUNCTION_TOKEN')]) {
-                    sh '''
+                    if (isUnix()) {
+                        sh '''
 curl -i -H 'Content-Type: application/json' -d '{"build_url":"'$BUILD_URL'"}' "https://jenkins-community-functions.azurewebsites.net/api/incrementals-publisher?clientId=default&code=$FUNCTION_TOKEN" || echo 'Problem calling Incrementals deployment function'
-                    '''
+                        '''
+                    } else {
+                        bat '''
+curl.exe -i -H 'Content-Type: application/json' -d '{"build_url":"'%BUILD_URL%'"}' "https://jenkins-community-functions.azurewebsites.net/api/incrementals-publisher?clientId=default&code=%FUNCTION_TOKEN%" || echo 'Problem calling Incrementals deployment function'
```
Comment (jglick) [inline vars/infra.groovy:270]: Does `'` work in batch scripts? Did you try this command with a `-v` and a phony token environment variable?

Comment (jetersen) [reply vars/infra.groovy:270]: I tested locally that the current string should work

Comment (jglick) [reply vars/infra.groovy:270]: https://ci.jenkins.io/job/Plugins/job/azure-credentials-plugin/job/master/35/execution/node/102/log/ suggests this is broken:

```

 FailRequestError: The incrementals-publisher invocation was missing the build_url attribute
     at IncrementalsPlugin.main (D:\home\site\wwwroot\incrementals-publisher\index.js:133:13)
     at module.exports (D:\home\site\wwwroot\incrementals-publisher\index.js:266:29)
     at WorkerChannel.invocationRequest (D:\Program Files (x86)\SiteExtensions\Functions\3.0.13139\64bit\workers\node\worker-bundle.js:17928:26)
     at ClientDuplexStream.WorkerChannel.eventStream.on (D:\Program Files (x86)\SiteExtensions\Functions\3.0.13139\64bit\workers\node\worker-bundle.js:17738:30)
     at ClientDuplexStream.emit (events.js:198:13)
     at addChunk (_stream_readable.js:288:12)
     at readableAddChunk (_stream_readable.js:269:11)
     at ClientDuplexStream.Readable.push (_stream_readable.js:224:10)
     at Object.onReceiveMessage (D:\Program Files (x86)\SiteExtensions\Functions\3.0.13139\64bit\workers\node\worker-bundle.js:21909:19)
     at InterceptingListener.module.exports.InterceptingListener.recvMessageWithContext (D:\Program Files (x86)\SiteExtensions\Functions\3.0.13139\64bit\workers\node\worker-bundle.js:21224:19)
     at D:\Program Files (x86)\SiteExtensions\Functions\3.0.13139\64bit\workers\node\worker-bundle.js:21323:14
```

CC @jetersen @olblak

Comment (jetersen) [reply vars/infra.groovy:270]: I suspect this was a copy paste error on my part.
```suggestion
curl.exe -i -H 'Content-Type: application/json' -d '{"build_url":"%BUILD_URL%"}' "https://jenkins-community-functions.azurewebsites.net/api/incrementals-publisher?clientId=default&code=%FUNCTION_TOKEN%" || echo 'Problem calling Incrementals deployment function'
```

Comment (jetersen) [reply vars/infra.groovy:270]: After actually testing it against site. I noticed a few things such as encoding being URL encoded...

```suggestion
curl.exe -i -H "Content-Type: application/json" -d "{""build_url"":""%BUILD_URL%""}" "https://jenkins-community-functions.azurewebsites.net/api/incrementals-publisher?clientId=default&code=%FUNCTION_TOKEN%" || echo "Problem calling Incrementals deployment function"
```

Comment (jetersen) [reply vars/infra.groovy:270]: https://stackoverflow.com/questions/562038/escaping-double-quotes-in-batch-script

**Pull request merged**